lms/djangoapps/teams/tests/test_views.py

```diff
@@ -1470,6 +1654,65 @@ def test_access_by_username(self, user, status, has_content):
             else:
                 self.assertEqual(membership['count'], 0)
 
+    @ddt.data(
+        ('student_masters', True),
+        ('student_masters_not_on_team', True),
+        ('student_unenrolled', False),
+        ('student_enrolled', True),
+        ('student_enrolled_both_courses_other_team', True),
+        ('staff', True),
+    )
+    @ddt.unpack
+    def test_access_by_username_organization_protected(self, user, can_see_bubble_team):
+        membership = self.get_membership_list(200, {'username': 'student_masters'}, user=user)
+        if can_see_bubble_team:
+            self.assertEqual(membership['count'], 1)
+            self.assertEqual(membership['results'][0]['team']['team_id'], self.masters_only_team.team_id)
+        else:
+            self.assertEqual(membership['count'], 0)
+
+    @ddt.unpack
+    @ddt.data(
+        ('student_on_team_1_private_set_1', True, True),
+        ('student_unenrolled', False, False),
+        ('student_enrolled', True, True),
+        ('student_on_team_2_private_set_1', True, True),
+        ('student_masters', True, True),
+        ('staff', True, True)
+    )
+    def test_access_by_username_private_teamset(self, user, can_see_any_teams, can_see_private_team):
+        self.masters_only_team.add_user(self.users['student_on_team_1_private_set_1'])
+        memberships = self.get_membership_list(200, {'username': 'student_on_team_1_private_set_1'}, user=user)
+        team_ids = [membership['team']['team_id'] for membership in memberships['results']]
+        if can_see_private_team:
+            self.assertEqual(len(team_ids), 2)
+            self.assertIn(self.team_1_in_private_teamset_1.team_id, team_ids)
+            self.assertIn(self.masters_only_team.team_id, team_ids)
+        elif can_see_any_teams:
+            self.assertEqual(len(team_ids), 1)
+            self.assertIn(self.masters_only_team.team_id, team_ids)
+        else:
+            self.assertEqual(len(team_ids), 0)
+
+    @ddt.unpack
+    @ddt.data(
+        ('student_on_team_1_private_set_1', 200),
+        ('student_unenrolled', 404),
+        ('student_enrolled', 403),
+        ('student_on_team_2_private_set_1', 403),
+        ('student_masters', 403),
+        ('staff', 200)
+    )
+    def test_access_by_team_private_teamset(self, user, expected_response):
+        memberships = self.get_membership_list(
```

Comment:
people aren't getting membership info, which is good and correct, but they're still getting the fact that the team exists. make these 404s.